crates/astria-conductor/src/driver.rs

```diff
@@ -170,36 +193,32 @@ impl Driver {
         Ok(())
     }
 
-    async fn handle_network_event(&self, event: NetworkEvent) -> Result<()> {
-        match event {
-            NetworkEvent::NewListenAddr(addr) => {
-                info!("listening on {}", addr);
-            }
-            NetworkEvent::GossipsubMessage(msg) => {
-                debug!("received gossip message");
-                let block = SequencerBlockData::from_bytes(&msg.data)
-                    .wrap_err("failed to deserialize SequencerBlockData received from network")?;
-
-                // validate block received from gossip network
-                self.block_verifier
-                    .validate_sequencer_block_data(&block)
-                    .await
-                    .wrap_err("invalid block received from gossip network")?;
-
-                info!(
-                    height = block.header().height.value(),
-                    "sequencer block received from p2p network",
-                );
-                self.executor_tx
-                    .send(ExecutorCommand::BlockReceivedFromGossipNetwork {
-                        block: Box::new(block),
-                    })
-                    .wrap_err("failed to send SequencerBlockData from network to executor")?;
+    async fn handle_new_block(&self, block: Result<SequencerBlockData, NewBlockStreamError>) {
+        let block = match block {
+            Err(err) => {
+                warn!(err.msg = %err, err.cause = ?err, "encountered an error while receiving a new block from sequencer");
+                return;
             }
-            _ => debug!("received network event"),
+            Ok(new_block) => new_block,
+        };
+
+        if let Err(err) = self
+            .block_verifier
+            .validate_sequencer_block_data(&block)
```

Comment:
But then we need to have a different endpoint? Don't we? Because otherwise we'd get responses from the same endpoint, which seems like it would defeat the purpose?

Reply:
yeah true, i guess we'd need to sync the validator set from genesis to validate this properly. we can probably remove for now then and just put in the docs that the sequencer node used by conductor is fully trusted

Reply:
Removed the logic and shuffled a few block_verifier bits around because its are now only used from `Reader`.